common/src/main/java/com/discordsrv/common/config/main/sync/MuteSyncConfig.java

```diff
@@ -0,0 +1,122 @@
+    public MinecraftToDiscordConfig minecraftToDiscord = new MinecraftToDiscordConfig();
+
+    @Comment("Options for syncing mutes from Discord to Minecraft")
+    public DiscordToMinecraftConfig discordToMinecraft = new DiscordToMinecraftConfig();
+
+    @Override
+    public boolean isSet() {
+        return serverId != 0;
+    }
+
+    @Override
+    public Game gameId() {
+        return Game.INSTANCE;
+    }
+
+    @Override
+    public Long discordId() {
+        return serverId;
+    }
+
+    @Override
+    public boolean isSameAs(MuteSyncConfig config) {
+        return false;
+    }
+
+    @Override
+    public String describe() {
+        return Long.toUnsignedString(serverId);
+    }
+
+    @ConfigSerializable
+    public static class DiscordToMinecraftConfig {
+
+        @Comment("What action(s) on Discord should trigger a mute/unmute in Minecraft. Valid options:\n"
+                + "timeout:  A timeout/untimeout on the Discord Server\n"
+                + "role:     Addition/removal of the muted role (specified above) to the user on Discord\n"
+                + "either:   Either of the above\n"
+                + "BEWARE: Settings of 'role' or 'either' can be exploited to remove mutes from players if %1 is set to 'discord'")
+        @Constants.Comment("tie-breaker")
+        public MuteSyncDiscordTrigger trigger = MuteSyncDiscordTrigger.TIMEOUT;
+
+        @Comment("The reason used when creating new mutes in Minecraft")
+        public String muteReasonFormat = "Muted on Discord: %punishment_reason%";
+
+        @Comment("The punisher shown when creating new mutes in Minecraft")
+        public String punisherFormat = "%user_color%@%user_name%";
+
+        @Comment("The message sent to the online player when they are muted in Minecraft due to a Discord mute")
+        public String muteNotificationMessage = "&cYou have been muted on Discord for &f%punishment_reason|'Unknown'% &cby &f%punishment_punisher%";
+
+        @Comment("The message sent to the online player when they are unmuted in Minecraft due to a Discord unmute")
+        public String unmuteNotificationMessage = "&aYou have been unmuted on Discord.";
+    }
+
+    @ConfigSerializable
+    public static class MinecraftToDiscordConfig {
+
+        @Comment("What action(s) to perform on the linked Discord account when a player is muted in Minecraft. Can be configured in more detail below\n"
+                + "Valid options:\n"
+                + "timeout:  Apply a Discord timeout on the user. (Must be 28 days or less)\n"
+                + "role:     Add the muted role (specified above) to the user on Discord"
+        )
+        public MuteSyncDiscordAction action = MuteSyncDiscordAction.TIMEOUT;
+
+        @Comment("When a mute duration is longer than 28 days (the maximum allowed in Discord), the muted role will be assigned instead of a timeout in such cases.\n"
+                + "Otherwise, no action will be taken on Discord for mutes longer than 28 days."
```

Comment:
Another option would be to have a scheduled task that periodically adds more days to the timeout, some thoughts:
- Worse UX for the Discord user
- Potentially better than granting a role (timeouts *might* block some other things on Discord side, such as dm:ing server members?)
- Lots more work to implement

Up to you, if you want to test or even implement it. Just thought I'd throw in some of my own thoughts